src/components/CheckBox/CheckBox.scss

```diff
@@ -19,106 +22,122 @@
   .ms-Label {
     font-size: $ms-font-size-m;
     padding: 0 0 0 26px;
+    cursor: pointer;
+    display: inline-block;
+  }
+}
+
+// The hidden input
+.ms-CheckBox-input {
+  position: absolute;
+  opacity: 0;
+}
+
+// The checkbox square container
+.ms-CheckBox-field::before {
+  content: '';
+  display: inline-block;
+  border: 2px solid $ms-color-neutralTertiary;
+  width: $ms-checkbox-field-size;
+  height: $ms-checkbox-field-size;
+  cursor: pointer;
+  font-weight: normal;
+  position: absolute;
+  box-sizing: border-box;
+  transition-property: background, border, border-color;
+  transition-duration: $ms-checkbox-transition-duration;
+  transition-timing-function: $ms-checkbox-transition-timing;
+}
+
+// The check mark icon
+.ms-CheckBox-field::after {
+  content: '\E73E';
```

Comment:
We should be using the icon mixins here instead of hard-coded Unicode values, e.g. `@include ms-Icon--CheckMark`